src/library/scala/collection/immutable/TreeMap.scala

```diff
@@ -27,6 +29,37 @@ object TreeMap extends ImmutableSortedMapFactory[TreeMap] {
   def empty[A, B](implicit ord: Ordering[A]) = new TreeMap[A, B]()(ord)
   /** $sortedMapCanBuildFromInfo */
   implicit def canBuildFrom[A, B](implicit ord: Ordering[A]): CanBuildFrom[Coll, (A, B), TreeMap[A, B]] = new SortedMapCanBuildFrom[A, B]
+  private val legacySerialisation = System.getProperty("scala.collection.immutable.TreeMap.newSerialisation", "false") != "false"
+
+  private class TreeMapProxy[A, B](
+    @transient private[this] var tree: RedBlackTree.Tree[A, B],
+    @transient private[this] var ordering: Ordering[A]) extends Serializable {
+
+    @throws[IOException]
+    private[this] def writeObject(out: java.io.ObjectOutputStream) = {
+      out.writeInt(RB.count(tree))
+      out.writeObject(ordering)
+      RB.foreachEntry(tree, {
+        (k: A, v: B) =>
+          out.writeObject(k)
+          out.writeObject(v)
+      })
+    }
+    @throws[IOException]
+    private[this] def readObject(in: java.io.ObjectInputStream) = {
+      val size = in.readInt()
+      ordering = in.readObject().asInstanceOf[Ordering[A]]
+      implicit val ord = ordering
+      for (i <- 0 until size) {
+        val key = in.readObject().asInstanceOf[A]
+        val value = in.readObject().asInstanceOf[B]
+        RB.update(tree, key, value, overwrite = true)
```

Comment:
You probably meant to assign this back to `tree`. At least this would explain why you're getting empty maps back :-)

Reply:
Doh

Reply:
updated to fix that and a couple of @transient that lagged over